# This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:
#
# Copyright (c) 2025 Qualcomm Technologies, Inc. All rights reserved.
#
# SPDX-License-Identifier: BSD-3-Clause-Clear

From 217538f0ffe4469f2b5261bf4adece541b873b77 Mon Sep 17 00:00:00 2001
From: Rafal Skierkowski <rskierko@qti.qualcomm.com>
Date: Thu, 21 Mar 2024 16:04:26 +0100
Subject: [PATCH 2/2] Skip device tree memory node modification.

Always skip device tree memory node modification. Memory nodes are
statically defined in Linux guest device tree.
In case outer device tree type is used and bootloader is passing device
tree to Linux VM guest such modification lead to boot problem.
Memory node modification is not needed whenever outer device tree
type is used or not, so it can be disabled.
This change might have impact on native boot.

Signed-off-by: Ankurkumar Satasiya <asatasiy@qti.qualcomm.com>
---
 .../edk2/QcomModulePkg/Library/BootLib/UpdateDeviceTree.c    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/bootloader/edk2/QcomModulePkg/Library/BootLib/UpdateDeviceTree.c b/bootloader/edk2/QcomModulePkg/Library/BootLib/UpdateDeviceTree.c
index 8794bda..a1d2122 100755
--- a/QcomModulePkg/Library/BootLib/UpdateDeviceTree.c
+++ b/QcomModulePkg/Library/BootLib/UpdateDeviceTree.c
@@ -344,6 +344,9 @@ AddMemMap (VOID *Fdt, UINT32 MemNodeOffset, BOOLEAN BootWith32Bit)
   UINT32 i = 0;
   UINT32 MemoryCellLen = 0;
 
+  DEBUG ((EFI_D_INFO, "Skip memory node modification\n"));
+  return EFI_SUCCESS;
+
   Status = QueryMemoryCellSize (Fdt, &MemoryCellLen);
   if (EFI_ERROR (Status)) {
     DEBUG ((EFI_D_ERROR, "ERROR: Not a valid memory node found!\n"));
@@ -879,4 +882,4 @@ DisableNonBootDeviceNode (VOID *fdt)
   }
   return Status;
 }
-#endif
\ No newline at end of file
+#endif
-- 
2.43.2

